flake8_fine_pytest/watchers/unique_test_names.py

```diff
@@ -0,0 +1,35 @@
+import ast
+from typing import Set
+
+from flake8_fine_pytest.watchers.base import BaseWatcher
+
+
+class UniqueTestNamesWatcher(BaseWatcher):
+    error_template = 'FP009 Duplicate name test case ({})'
+
+    def run(self) -> None:
+        testcases_names: Set[str] = set()
+        duplicates: Set[str] = set()
+
+        if not self._is_test_file(self.filename):
+            return
+
+        for node in ast.walk(self.tree):
+            if not self._is_properly_node(node):
+                continue
+
+            node_name = node.name  # type: ignore
+
+            if node_name not in testcases_names:
+                testcases_names.add(node_name)
+                continue
+
+            if node_name in duplicates:
+                continue
+
+            duplicates.add(node_name)
+            error_msg = self.error_template.format(node_name)
+            self.add_error((node.lineno, 0, error_msg))
+
+    def _is_properly_node(self, node: ast.AST) -> bool:
```

Comment:
This is copy paste. We use this method already in 3 places and 2 of them have similar code. I think we should move this method to `BaseWatcher`, make it public, and override where we need other logic (like [here](https://github.com/best-doctor/flake8-fine-pytest/blob/9eb2b9f4e7029deac9d33027cbb2bba1d992b938/flake8_fine_pytest/watchers/xfail_until_argument_watcher.py#L72))

Reply:
By the way, we have similar case in [another PR](https://github.com/best-doctor/flake8-fine-pytest/pull/8/files#r529247014). Please, take a look, maybe we will do it earlier.

Reply:
Fixed https://github.com/best-doctor/flake8-fine-pytest/pull/7/commits/35f68b6dca164bf37eacfcd1be458d86c6b142cf

Reply:
Have a [comment](https://github.com/best-doctor/flake8-fine-pytest/pull/7/files#r532197073)